extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/KubernetesClientProcessor.java

```diff
@@ -23,12 +30,25 @@
 import io.quarkus.jackson.deployment.IgnoreJsonDeserializeClassBuildItem;
 import io.quarkus.kubernetes.client.runtime.KubernetesClientProducer;
 import io.quarkus.kubernetes.spi.KubernetesRoleBuildItem;
+import io.quarkus.runtime.annotations.RegisterForReflection;
 
 public class KubernetesClientProcessor {
 
     private static final DotName WATCHER = DotName.createSimple("io.fabric8.kubernetes.client.Watcher");
+    private static final DotName RESOURCE_EVENT_HANDLER = DotName
+            .createSimple("io.fabric8.kubernetes.client.informers.ResourceEventHandler");
     private static final DotName KUBERNETES_RESOURCE = DotName
             .createSimple("io.fabric8.kubernetes.api.model.KubernetesResource");
+    private static final DotName KUBERNETES_RESOURCE_LIST = DotName
+            .createSimple("io.fabric8.kubernetes.api.model.KubernetesResourceList");
+    private static final Set<String> EXCLUDED_CLASSES;
```

Comment:
I think that we'll need a better name for this

Reply:
Sure, do you have suggestions?

Reply:
Maybe `EXCLUDED_WATCH_CLASSES`?